src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -22,34 +22,30 @@ CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 source "${CURRENT_DIR}/../integration_test_setup.sh" \
   || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
 
+# Check if all the tools required by CC coverage are installed.
+[[ ! -x /usr/bin/lcov ]] && fail "lcov not installed. Skipping test."
+[[ -z $( which gcov ) ]] && fail "gcov not installed."
+[[ -z $( which g++ ) ]] && fail "g++ not installed."
+
+# These are the variables needed by tools/test/collect_cc_coverage.sh
+# They will be properly sub-shelled when invoking the script.
+readonly coverage_dir="${PWD}"
```

Comment:
Use all-caps as this is a constant (https://google.github.io/styleguide/shell.xml?showone=Constants_and_Environment_Variable_Names#Constants_and_Environment_Variable_Names). You can still do X=$X in the subshelling.

Reply:
I cannot do X=$X if they are readonly. I renamed them with a suffix.